paymentsheet/src/main/java/com/stripe/android/paymentsheet/IntentConfirmationHandler.kt

```diff
@@ -0,0 +1,103 @@
+package com.stripe.android.paymentsheet
+
+import android.app.Activity
+import androidx.activity.result.ActivityResultCaller
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleOwner
+import com.stripe.android.core.strings.ResolvableString
+import com.stripe.android.model.StripeIntent
+import com.stripe.android.paymentsheet.addresselement.AddressDetails
+import com.stripe.android.paymentsheet.model.PaymentSelection
+import kotlinx.coroutines.flow.StateFlow
+
+/**
+ * This interface handles the process of confirming a [StripeIntent]. This interface can only handle confirming one
+ * intent at a time.
+ */
+internal interface IntentConfirmationHandler {
+    /**
+     * A flow that provides the state of [IntentConfirmationHandler]
+     */
+    val state: StateFlow<State>
```

Comment:
I ended up going back on separating the `state` and `result` flows. It ended up being too awkward attempting to manage the two when handling the primary button & wallet loading logic.